ext/node/polyfills/os.ts

```diff
@@ -315,12 +316,28 @@ export function uptime(): number {
   return osUptime();
 }
 
-/** Not yet implemented */
+/** partially implemented
+ * https://nodejs.org/api/os.html#osuserinfooptions
+ * missing gid, uid
+ */
 export function userInfo(
-  // deno-lint-ignore no-unused-vars
   options: UserInfoOptions = { encoding: "utf-8" },
 ): UserInfo {
-  notImplemented(SEE_GITHUB_ISSUE);
+  if (options.encoding !== "utf-8") notImplemented(SEE_GITHUB_ISSUE);
+
+  const username = Deno.env.get("USER");
+  const homedir = Deno.env.get("HOME");
+  const shell = Deno.env.get("SHELL") || null;
+
+  if (!username || !homedir) throw codes.SYSERR;
```

Comment:
should I throw `notImplemented(SEE_GITHUB_ISSUE);` here too? node does a system error when those aren't set